src/main/java/org/dice_research/opal/launuts/ModelBuilder.java

```diff
@@ -43,9 +56,93 @@
 		// Additional prefixes to reduce model size
 		model.setNsPrefix("laude", Vocabularies.NS_LAU_DE);
 		model.setNsPrefix("nutscode", Vocabularies.NS_EU_NUTS_CODE);
+
+		// Prefix for identifying a geometry type
+		model.setNsPrefix("sf", "http://www.opengis.net/ont/sf#");
+	}
+
+	public void addPolygons(Resource res, String resource_id, JSONArray nuts_or_lau_polygons, String resource_id_type) {
```

Comment:
In this function addPolygons,  multiple nested loops and conditional statements are present, which increases the overall run time complexity